### ISSUE TITLE
False positive in nonportable_path_linter

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1356]
As seen in https://github.com/microsoft/LightGBM/pull/5249

```r
grepl(
  pattern = "[Ordinal/Name Pointer] Table"
  pattern = "[Ordinal/Name Pointer] Table"
  , x = objdump_results
  , fixed = TRUE
)
# <text>:2:14: warning: Use file.path() to construct portable file paths.
#   pattern = "[Ordinal/Name Pointer] Table"
#              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
# <text>:3:3: error: unexpected symbol
#   pattern = "[Ordinal/Name Pointer] Table"
#   ^
```

**Comment by IndrajeetPatil** [comment]
A more minimal test case example:

``` r
lintr::lint(
  text = "grepl('[a/b] c', x, fixed = TRUE)",
  linters = lintr::nonportable_path_linter()
)
#> <text>:1:8: warning: [nonportable_path_linter] Use file.path() to construct portable file paths.
#> grepl('[a/b] c', x, fixed = TRUE)
#>        ^~~~~~~
```

<sup>Created on 2022-07-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by IndrajeetPatil** [comment]
Root cause of the problem, since this should return `FALSE` here:

``` r
lintr:::is_path("[a/b] c")
#> [1] TRUE
```

<sup>Created on 2022-07-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

This regex needs to be modified 🙈

``` r
lintr:::path_regex
#> (?:(?:(?:^/(?![[:space:]/]))|(?:^~(?:[A-Za-z0-9_\-.])*(?:/)*(?![[:space:]'"]))|(?:^[[:alpha:]]:(?:[/\\])?(?![[:space:]/\\]))|(?:^\\\\(?:(?:[A-Za-z0-9_\-.])+(?:\\)?)?(?![[:space:]\\'"])))|(?:^(?!(?:['"]|(?:(?:^/(?![[:space:]/]))|(?:^~(?:[A-Za-z0-9_\-.])*(?:/)*(?![[:space:]'"]))|(?:^[[:alpha:]]:(?:[/\\])?(?![[:space:]/\\]))|(?:^\\\\(?:(?:[A-Za-z0-9_\-.])+(?:\\)?)?(?![[:space:]\\'"])))|(?:[[:alpha:]])+://)))(?:(?:(?:(?:(?:[[:alnum:]]|
#> |[!#$%&'()+,\-.;=@\[\]\^_`{}~]))+(?:/|\\))|(?:\.(?:\.)?$))(?!['"])))
```

<sup>Created on 2022-07-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by bahadzie** [comment]
Hi. I've hit a similar false positive using `grepl`. Is this something I can help out with? I'm relatively new to R but I'm willing to give it a shot.
